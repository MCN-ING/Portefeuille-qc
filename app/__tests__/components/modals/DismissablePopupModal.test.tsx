import { testIdWithKey } from '@hyperledger/aries-bifold-core'
import { render, fireEvent } from '@testing-library/react-native'
import React from 'react'

import DismissiblePopupModal from '../../../src/components/modals/DismissablePopupModal'

const title = 'Test Title'
const description = 'Lorem ipsum sit dolar amet'

describe('DismissiblePopupModal Component', () => {
  test('Renders correctly without call to action', () => {
    const onDismissPressed = jest.fn()
    const tree = render(
      <DismissiblePopupModal title={title} description={description} onDismissPressed={onDismissPressed} />
    )

    expect(tree).toMatchSnapshot()
  })

  test('Renders correctly with call to action', () => {
    const onDismissPressed = jest.fn()
    const onCallToActionPressed = jest.fn()
    const tree = render(
      <DismissiblePopupModal
        title={title}
        description={description}
        onDismissPressed={onDismissPressed}
        onCallToActionPressed={onCallToActionPressed}
      />
    )

    expect(tree).toMatchSnapshot()
  })

  test('Triggers call to action on press', () => {
    const onDismissPressed = jest.fn()
    const onCallToActionPressed = jest.fn()
    const { getByTestId } = render(
      <DismissiblePopupModal
        title={title}
        description={description}
        onDismissPressed={onDismissPressed}
        onCallToActionPressed={onCallToActionPressed}
      />
    )
    const okayButton = getByTestId(testIdWithKey('Okay'))
    fireEvent(okayButton, 'press')

    expect(onCallToActionPressed).toBeCalledTimes(1)
  })

  test('Triggers dismiss on press', () => {
    const onDismissPressed = jest.fn()
    const onCallToActionPressed = jest.fn()
    const { getByTestId } = render(
      <DismissiblePopupModal
        title={title}
        description={description}
        onDismissPressed={onDismissPressed}
        onCallToActionPressed={onCallToActionPressed}
      />
    )
    const closeButton = getByTestId(testIdWithKey('Close'))
    fireEvent(closeButton, 'press')

    expect(onDismissPressed).toBeCalledTimes(1)
  })
})
