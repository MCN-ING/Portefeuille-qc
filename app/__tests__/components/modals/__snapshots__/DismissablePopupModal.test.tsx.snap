// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DismissiblePopupModal Component Renders correctly with call to action 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={false}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "flex": 1,
          "justifyContent": "center",
          "minHeight": 1334,
          "minWidth": 750,
          "padding": 20,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessible={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#313132",
            "borderColor": "#0099FF",
            "borderRadius": 5,
            "borderWidth": 1,
            "flex": 1,
            "maxHeight": "50%",
            "minWidth": 700,
            "padding": 10,
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingHorizontal": 5,
              "paddingTop": 5,
            }
          }
        >
          <View
            style={
              {
                "alignSelf": "center",
                "marginRight": 10,
              }
            }
          >
            <Text
              accessible={false}
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#0099FF",
                    "fontSize": 30,
                  },
                  undefined,
                  {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              {
                "flexGrow": 1,
              }
            }
          >
            <Text
              style={
                {
                  "alignSelf": "flex-start",
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "bold",
                }
              }
              testID="com.ariesbifold:id/HeaderText"
            >
              Test Title
            </Text>
          </View>
          <View
            style={
              {
                "alignSelf": "flex-end",
              }
            }
          >
            <View
              accessibilityLabel="Global.Close"
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              hitSlop={
                {
                  "bottom": 44,
                  "left": 44,
                  "right": 44,
                  "top": 44,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
              testID="com.ariesbifold:id/Close"
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#0099FF",
                      "fontSize": 30,
                    },
                    undefined,
                    {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "flexShrink": 1,
              "justifyContent": "space-between",
              "marginLeft": 40,
              "paddingBottom": 5,
              "paddingHorizontal": 5,
            }
          }
        >
          <RCTScrollView
            contentContainerStyle={
              {
                "flexGrow": 1,
              }
            }
            showsVerticalScrollIndicator={false}
            style={
              {
                "flex": 1,
              }
            }
          >
            <View>
              <View
                onStartShouldSetResponder={[Function]}
              >
                <Text
                  selectable={true}
                  style={
                    {
                      "color": "#FFFFFF",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "paddingVertical": 16,
                    }
                  }
                  testID="com.ariesbifold:id/BodyText"
                >
                  Lorem ipsum sit dolar amet
                </Text>
              </View>
            </View>
          </RCTScrollView>
          <View
            style={
              {
                "paddingTop": 10,
              }
            }
          >
            <View
              accessibilityLabel="Global.Okay"
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#42803E",
                  "borderRadius": 4,
                  "opacity": 1,
                  "padding": 16,
                }
              }
              testID="com.ariesbifold:id/Okay"
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#FFFFFF",
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "textAlign": "center",
                      },
                      false,
                      false,
                      false,
                    ]
                  }
                >
                  Global.Okay
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`DismissiblePopupModal Component Renders correctly without call to action 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={false}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "flex": 1,
          "justifyContent": "center",
          "minHeight": 1334,
          "minWidth": 750,
          "padding": 20,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessible={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#313132",
            "borderColor": "#0099FF",
            "borderRadius": 5,
            "borderWidth": 1,
            "flex": 1,
            "maxHeight": "50%",
            "minWidth": 700,
            "padding": 10,
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingHorizontal": 5,
              "paddingTop": 5,
            }
          }
        >
          <View
            style={
              {
                "alignSelf": "center",
                "marginRight": 10,
              }
            }
          >
            <Text
              accessible={false}
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#0099FF",
                    "fontSize": 30,
                  },
                  undefined,
                  {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              {
                "flexGrow": 1,
              }
            }
          >
            <Text
              style={
                {
                  "alignSelf": "flex-start",
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "bold",
                }
              }
              testID="com.ariesbifold:id/HeaderText"
            >
              Test Title
            </Text>
          </View>
          <View
            style={
              {
                "alignSelf": "flex-end",
              }
            }
          >
            <View
              accessibilityLabel="Global.Close"
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              hitSlop={
                {
                  "bottom": 44,
                  "left": 44,
                  "right": 44,
                  "top": 44,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
              testID="com.ariesbifold:id/Close"
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#0099FF",
                      "fontSize": 30,
                    },
                    undefined,
                    {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "flexShrink": 1,
              "justifyContent": "space-between",
              "marginLeft": 40,
              "paddingBottom": 5,
              "paddingHorizontal": 5,
            }
          }
        >
          <RCTScrollView
            contentContainerStyle={
              {
                "flexGrow": 1,
              }
            }
            showsVerticalScrollIndicator={false}
            style={
              {
                "flex": 1,
              }
            }
          >
            <View>
              <View
                onStartShouldSetResponder={[Function]}
              >
                <Text
                  selectable={true}
                  style={
                    {
                      "color": "#FFFFFF",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "paddingVertical": 16,
                    }
                  }
                  testID="com.ariesbifold:id/BodyText"
                >
                  Lorem ipsum sit dolar amet
                </Text>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
